pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -287,6 +287,14 @@ func SetDefaults_Shoot(obj *Shoot) {
 
 		obj.Spec.Provider.Workers[i].CRI = &CRI{Name: CRINameContainerD}
 	}
+
+	if obj.Spec.SystemComponents == nil {
+		obj.Spec.SystemComponents = &SystemComponents{}
+	}
+	if obj.Spec.SystemComponents.CoreDNS == nil {
+		obj.Spec.SystemComponents.CoreDNS = &CoreDNS{}
+		obj.Spec.SystemComponents.CoreDNS.Autoscaling.Mode = CoreDNSAutoscalingModeHorizontal
```

Comment:
If I set `.spec.systemComponents.coreDNS={}` in my shoot then the autoscaling mode will not be defaulted.

Reply:
Adapted. I hope this is the right way to do it.